### ISSUE TITLE
Foreground accuracy to be 0.0

### ISSUE THREAD

**Issue by deepakkupanda** [open-mmlab/mmsegmentation#1641]
Hi, 
I am trying to train my custom dataset. I used two datasets such as P3M and DUTS for doing binary classifications. I have made all intensities of seg images to be in-between 0 and 1.

<img width="392" alt="image" src="https://user-images.githubusercontent.com/41481181/172111050-93caee1d-4b4d-46c7-b842-4cb826d96ec5.png">
<img width="413" alt="image" src="https://user-images.githubusercontent.com/41481181/172111179-5d51815c-5976-4985-b224-bc97d24dd9cc.png">

**Comment by deepakkupanda** [comment]
From the documents, I see
![image](https://user-images.githubusercontent.com/41481181/172111919-e583a7b1-f201-4e63-8958-9c0937d278e6.png)

**Comment by MengzhangLI** [comment]
Try to set num_classes = 2 in config.

**Comment by deepakkupanda** [comment]
I have kept the config classes to be 2.
<img width="299" alt="image" src="https://user-images.githubusercontent.com/41481181/172126758-211c4071-059b-4e63-bcdc-6050ddca6fb6.png">

I am also sharing the log file with you
[20220603_170927.log](https://github.com/open-mmlab/mmsegmentation/files/8842999/20220603_170927.log)

I used this to train my model.
tools/dist_train.sh configs/pspnet/pspnet_r101-d8_512x512_160k_p3m10k.py 1 --work-dir work_dirs/pspnet_r101-d8_512x512_160k_p3m10k/ --deterministic

**Comment by deepakkupanda** [comment]
@MengzhangLI  What is the root cause?

**Comment by MengzhangLI** [comment]
> @MengzhangLI What is the root cause?

Your dataset annotations have two types label: label 0, label 1 and you set `reduce_zero_label=True` and `num_class=2` at the same time.

Could you try to set `reduce_zero_label=False` and `num_class=2`?

**Comment by MengzhangLI** [comment]
This modification is preferred:
 (1) reduce_zero_label=False, (2) num_class=1, (3) use binary cross entropy loss (use_sigmoid=True) in config file.

**Comment by deepakkupanda** [comment]
Thank you @MengzhangLI  I will use this config.

**Comment by deepakkupanda** [comment]
Hi @MengzhangLI ,

Do I need to mention, type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)), in both decode head and auxillary head? 

Thanks,

**Comment by LansonChen** [comment]
@deepakkupanda  Hi, did u solve this problem?

**Comment by MeowZheng** [comment]
> Hi @MengzhangLI ,
> 
> Do I need to mention, type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0)), in both decode head and auxillary head?
> 
> Thanks,

yes
